zjb/templates/results.html.j2

```diff
@@ -36,4 +53,12 @@
                 </tfoot>
             </table>
             {%- endfor -%}
+
+            <script>
+                function openModalDialog(){
+                    const favDialog = document.getElementById('favDialog');
+                    favDialog.showModal();
+                }
```

Comment:
I am afraid this will not work as intended;
with code above, you create multiple elements with the same ID (which is invalid HTML code), then here in this script it will always fetch the first (or last, I am not sure) from HTML, not the closest.
The better option would be to add job id in the dialog ID and then add it as a parameter in the `openModalDialog()` function.